services/graph/pkg/service/v0/api_driveitem_permissions.go

```diff
@@ -283,7 +283,7 @@ func createShareRequestToFederatedUser(user libregraph.User, resourceId *storage
 			Id: &storageprovider.Grantee_UserId{UserId: &userpb.UserId{
 				Type:     userpb.UserType_USER_TYPE_FEDERATED,
 				OpaqueId: user.GetId(),
-				Idp:      providerInfo.Domain, // the domain is persisted in the grant as u:{opaqueid}:{domain}
+				Idp:      *user.GetIdentities()[0].Issuer, // the domain is persisted in the grant as u:{opaqueid}:{domain}
```

Comment:
Is it guaranteed that
A) `len(user.GetIdentities()) > 0`
B) `user.GetIdentities()[0] != nil`
? Otherwise we need checks for this because it would panic.

Reply:
yes there is a check above the only line making a call to this function:
```go
if len(user.Identities) < 1 {
    return libregraph.Permission{}, errorcode.New(errorcode.InvalidRequest, "user has no federated identity")
}
// [...]
createShareRequest := createShareRequestToFederatedUser(user, statResponse.GetInfo().GetId(), providerInfoResp.ProviderInfo, cs3ResourcePermissions)
```